src/Tests/Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests/ParameterCapturing/ParameterCapturingLoggerTests.cs

```diff
@@ -0,0 +1,102 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing;
+using Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.FunctionProbes;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests.ParameterCapturing
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    public class ParameterCapturingLoggerTests
+    {
+        private readonly MethodInfo _testMethod = typeof(ParameterCapturingLoggerTests).GetMethod(nameof(TestMethod), BindingFlags.Static | BindingFlags.NonPublic);
+        private static void TestMethod() { }
+
+        [Theory]
+        [InlineData(ParameterCaptureMode.Inline, typeof(DotnetMonitor.ParameterCapture.UserCode))]
+        [InlineData(ParameterCaptureMode.Background, typeof(DotnetMonitor.ParameterCapture.SystemCode))]
+        internal void LoggingCategories(ParameterCaptureMode mode, Type categoryType)
+        {
+            // Arrange & Act
+            IList<LogRecordEntry> entries = TestCore(mode);
+
+            // Assert
+            LogRecordEntry entry = Assert.Single(entries);
+            Assert.Equal(categoryType.FullName, entry.Category);
+
+            return;
+        }
+
+        [Theory]
+        [InlineData(ParameterCaptureMode.Inline)]
+        [InlineData(ParameterCaptureMode.Background)]
+        internal void ScopeData(ParameterCaptureMode mode)
+        {
+            // Arrange
+            using Activity loggingActivity = new("ScopeDataTest");
+            Activity.Current = loggingActivity;
+            loggingActivity.Start();
+
+            Dictionary<string, object> expectedScope = new()
+            {
+                { ParameterCapturingLogger.Scopes.ActivityId, loggingActivity.Id },
+                { ParameterCapturingLogger.Scopes.ActivityIdFormat, loggingActivity.IdFormat },
+                { ParameterCapturingLogger.Scopes.ThreadId, Environment.CurrentManagedThreadId },
+                { ParameterCapturingLogger.Scopes.CaptureSite.MethodName, _testMethod.Name },
+                { ParameterCapturingLogger.Scopes.CaptureSite.TypeName, _testMethod.DeclaringType.FullName },
+                { ParameterCapturingLogger.Scopes.CaptureSite.ModuleName, _testMethod.Module.Name }
+            };
+
+            // Act
+            IList<LogRecordEntry> entries = TestCore(mode);
+
+            // Assert
+            LogRecordEntry entry = Assert.Single(entries);
+            IReadOnlyList<KeyValuePair<string, object>> rawScope = Assert.Single(entry.Scopes);
+
+            Dictionary<string, object> scopeData = new(rawScope);
+
+            // Validate variant data first.
```

Comment:
The "better" way to do this is to replace the usage of `DateTime.Now` and the other "Now" variants with `TimeProvider`:
- .NET 8+: https://learn.microsoft.com/en-us/dotnet/api/system.timeprovider?view=net-8.0
- Custom shim for lower TFMs: https://github.com/dotnet/dotnet-monitor/blob/main/src/Tools/dotnet-monitor/TimeProvider.cs

You'd add a TimeProvider parameter to the `ParameterCapturingLogger` ctor. In product code, you'd pass `TimeProvider.System`. In test code, you'd pass your test-derived version that you can manually control the time.

I won't push you to do it here, since the time component is fairly simple, but this is mostly just a heads up. If the code evolves to be more complex than just snapshotting the timestamp, then I'd push for using `TimeProvider`.